classes/PrestaShopAutoload.php

```diff
@@ -57,11 +57,11 @@ class PrestaShopAutoload
 
     protected function __construct()
     {
-        $this->root_dir = _PS_CORE_DIR_ . '/';
+        $this->root_dir = dirname(__DIR__);
```

Comment:
```suggestion
        $this->root_dir = dirname(__DIR__) . '/';
```

This will avoid adding the slash everywhere else in this class

Reply:
However I don't understand `dirname(__DIR__)` is supposed to be equal to the `classes` folder Not the root folder so it seems incorrect unless I'm missing something

Reply:
> This will avoid adding the slash everywhere else in this class

that is correct, but the convenction is normally to not add the `/` to a dir variable, which is also safer, as user can set the path /wo the trailing slash.

Reply:
> However I don't understand `dirname(__DIR__)` is supposed to be equal to the `classes` folder Not the root folder so it seems incorrect unless I'm missing something

![image](https://user-images.githubusercontent.com/2228672/161578236-2b500aec-e5cb-4f90-a1e4-c25d89bca5aa.png)

`_PS_CORE_DIR_` is the PrestaShop/project root directory

Reply:
@jolelievre , I agree with your comment, it is disturbing cause `__DIR__` is current php file directory and `dirname` gives the parent directory of this directory xD
So this `dirname(__DIR__)` is like a `/..` assuming that current autoload file is at two levels from `_PS_CORE_DIR_` which is project root directory ...

:thinking: 
I don't like this change cause it is like changing configuration to hardcode.
I can accept that autoloading is based on an know directory tree for classes and I don't know the full Prestashop to understand if this change is good or not.

Is it the only autoload point ?
Has it was configuration and now it is hardcoded, what happens for other files that use `_PS_CORE_DIR_` ?
Do you assume that this variable isn't the same meaning than this usage for autoload file ?
(imagine we change the project structure for example, should we update several hardcoded files ?)